### DISCUSSION TITLE
Replace variables using slash commands

### DISCUSSION THREAD

**Discussion by SDGLBL** [olimorris/codecompanion.nvim#115] [Ideas]
Currently, there is an issue with #buffer #buffers #lsp where users cannot see the specific content represented by these symbols after introducing these commands, nor can they modify this content. Referring to zed, I think we can use nvim-cmp to introduce a similar slash command that allows users to quickly bring context into the chat buffer using /cmd args. This actually eliminates the complex variable replacement mechanism in chat implementation and feels more intuitive overall. At present, I have implemented an initial version of this feature through nvim-cmp, and it works very smoothly.

![CleanShot 2024-08-25 at 00 25 05](https://github.com/user-attachments/assets/3705ed12-7a9e-496c-aaa3-4464a5e5badd)

**Comment by olimorris** [comment]
You technically can see it if you open up debug mode in the chat buffer with `gd`. But yep it's very clunky right now. But this will be better once I push the messages refactor through later this weekend.

**Reply by olimorris** [reply]
Could you share this in a PR? This looks awesome

**Reply by SDGLBL** [reply]
@olimorris  https://github.com/olimorris/codecompanion.nvim/pull/119
